internal/querynodev2/services.go

```diff
@@ -236,13 +236,8 @@ func (node *QueryNode) WatchDmChannels(ctx context.Context, req *querypb.WatchDm
 		return merr.Success(), nil
 	}
 
-	node.manager.Collection.PutOrRef(req.GetCollectionID(), req.GetSchema(),
+	node.manager.Collection.Put(req.GetCollectionID(), req.GetSchema(),
```

Comment:
if watch channel failed, need to call `Release` with defer to avoid leak

Reply:
nice catch